### PR TITLE
Add new rule GenericTypesNames

### PR DESCRIPTION
**Pull request by su8898**
More info: https://twitter.com/dsymetweets/status/1235330491311677440

### CODE REVIEW

File: src/FSharpLint.Core/fsharplint.json [hunk 1]
```diff
@@ -201,11 +201,18 @@
...
+        "enabled": true,
+        "config": {
+            "naming": "PascalCase",  
+            "underscores": "None"
+        }
     },
     "publicValuesNames": {
         "enabled": true,
         "config": {
-            "underscores": "AllowPrefix"
```
Comment (knocte) [inline src/FSharpLint.Core/fsharplint.json:208]: @su8898 this change here is unrelated, can you drop it from the PR please?

Comment (su8898) [reply src/FSharpLint.Core/fsharplint.json]: @knocte I've removed the misalignment and updated the PR. Could you please have a look again? thank you.

**Comment by duckmatt** [comment]
Interesting with the mac os build issue, I'll try to have a look at bringing the mac os build back tomorrow when I'm at a computer again

**Comment by knocte** [comment]
Thanks for merging Matt.

> Interesting with the mac os build issue, I'll try to have a look at bringing the mac os build back tomorrow when I'm at a computer again

I think I've found a way to fix the mac os build, will open PR today.
